miasm2/expression/expression.py

```diff
@@ -344,15 +343,21 @@ class ExprInt(Expr):
      - Constant 0x12345678 on 32bits
      """
 
-    def __init__(self, arg):
-        """Create an ExprInt from a numpy int
-        @arg: numpy int"""
-
-        if not is_modint(arg):
-            raise ValueError('arg must by numpy int! %s' % arg)
-
-        self._arg = arg
-        self._size = self.arg.size
+    def __init__(self, arg, size=None):
+        """Create an ExprInt from a modint or int/size
+        @arg: modint int or int
+        @size: (optionnal) int size"""
+
+        if is_modint(arg):
+            self._arg = arg
+            self._size = self.arg.size
+            if size is not None:
+                assert(arg.size == size)
```

Comment:
You could maybe add an error message for the user to understand the error easily